perception/detected_object_validation/src/object_lanelet_filter.cpp

```diff
@@ -23,6 +24,59 @@
 
 #include <lanelet2_core/geometry/Polygon.h>
 
+namespace
+{
+class TransformFootprintPoint
+{
+private:
+  tf2::Transform tf_target2objects_world_;
+  tf2::Transform tf_target2objects_;
+  tf2::Transform tf_objects_world2objects_;
+  geometry_msgs::msg::TransformStamped tf_target2objects_stamped_;
+
+public:
+  bool setTransformTarget2ObjectsWorld(
+    const autoware_auto_perception_msgs::msg::DetectedObjects &, const std::string &,
+    const tf2_ros::Buffer &);
+  bool setTransformTarget2Objects(const geometry_msgs::msg::Pose &);
+  geometry_msgs::msg::Point32 transformPoint(const geometry_msgs::msg::Point32 &);
+};
+
+bool TransformFootprintPoint::setTransformTarget2ObjectsWorld(
+  const autoware_auto_perception_msgs::msg::DetectedObjects & input_msg,
+  const std::string & target_frame_id, const tf2_ros::Buffer & tf_buffer)
+{
+  const auto ros_target2objects_world =
+    getTransform(tf_buffer, input_msg.header.frame_id, target_frame_id, input_msg.header.stamp);
+  if (!ros_target2objects_world) {
+    return false;
+  }
+  tf2::fromMsg(*ros_target2objects_world, tf_target2objects_world_);
+
+  return true;
+}
+
+bool TransformFootprintPoint::setTransformTarget2Objects(const geometry_msgs::msg::Pose & pose)
+{
+  // get tf object to map
+  tf2::fromMsg(pose, tf_objects_world2objects_);
+  tf_target2objects_ = tf_target2objects_world_ * tf_objects_world2objects_;
+  tf_target2objects_stamped_.transform = tf2::toMsg(tf_target2objects_);
+
+  return true;
+}
+
+geometry_msgs::msg::Point32 TransformFootprintPoint::transformPoint(
+  const geometry_msgs::msg::Point32 & point)
+{
+  geometry_msgs::msg::Point32 point_transformed;
+  tf2::doTransform<geometry_msgs::msg::Point32>(
+    point, point_transformed, tf_target2objects_stamped_);
+
+  return point_transformed;
+}
+}  // namespace
+
```

Comment:
I think no need to separate as the class.
Can you use [this](https://github.com/autowarefoundation/autoware.universe/blob/main/common/perception_utils/include/perception_utils/perception_utils.hpp#L199-L228)?

Reply:
It is difficult to use this.
That is because
- 1. I want to use `tf_target2objects_`, but [`transformObjects`](https://github.com/autowarefoundation/autoware.universe/blob/ce93396d7b17dbb37666a7251d148fa1dc26aa4d/common/perception_utils/include/perception_utils/perception_utils.hpp#L199-L228) does not return tf.
- 2. I want to use transformFootprintPoint in for-loop(objects) and for-loop(points). However, [`transformObjects`](https://github.com/autowarefoundation/autoware.universe/blob/ce93396d7b17dbb37666a7251d148fa1dc26aa4d/common/perception_utils/include/perception_utils/perception_utils.hpp#L199-L228) includes for-loop(objects). If I call [`transformObjects`](https://github.com/autowarefoundation/autoware.universe/blob/ce93396d7b17dbb37666a7251d148fa1dc26aa4d/common/perception_utils/include/perception_utils/perception_utils.hpp#L199-L228) in for-loop(points), it is not efficient.

Reply:
@yukkysaito
I used transformPoint and cared the quaternion in https://github.com/autowarefoundation/autoware.universe/commit/35fa236fb5c5c2e99368165d4adf01fabfa16bb7